api.go

```diff
@@ -519,15 +519,21 @@ func (c Client) executeMethod(method string, metadata requestMetadata) (res *htt
 	// Indicate to our routine to exit cleanly upon return.
 	defer close(doneCh)
 
+	// Retry only when reader is seekable
+	reqRetry := MaxRetry
+	if !isSeekable {
+		reqRetry = 1
+	}
+
 	// Blank indentifier is kept here on purpose since 'range' without
 	// blank identifiers is only supported since go1.4
 	// https://golang.org/doc/go1.4#forrange.
-	for range c.newRetryTimer(MaxRetry, DefaultRetryUnit, DefaultRetryCap, MaxJitter, doneCh) {
+	for range c.newRetryTimer(reqRetry, DefaultRetryUnit, DefaultRetryCap, MaxJitter, doneCh) {
 		// Retry executes the following function body if request has an
 		// error until maxRetries have been exhausted, retry attempts are
 		// performed after waiting for a given period of time in a
 		// binomial fashion.
-		if isRetryable {
+		if isSeekable {
```

Comment:
Why change the variable name? @vadmeste `isRetryable` is fine here. isSeekable is not providing any meaning.

Reply:
> Why change the variable name? @vadmeste isRetryable is fine here. isSeekable is not providing any meaning. Its also not associated with what this PR tries to fix.

Yeah, it is a simple change, I thought isSeekable is more precise than isRetryable, will revert